tmk_core/common/backlight.h

```diff
@@ -23,8 +23,9 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 typedef union {
     uint8_t raw;
     struct {
-        bool    enable :1;
-        uint8_t level  :7;
+        bool    enable    :1;
+        bool    breathing :1;
```

Comment:
Could you put the breathing at the end, so it doesn't wipe out anyone settings.  

Also .... it may be worth setting level to use 4 bits, rather than 6 or 7.   If the max is 15, then 4 bits is all we'd really need here.

Reply:
I ran `egrep` on current master for `BACKLIGHT_LEVELS` in `keyboards/`, `users/` and `layouts/`, then filtered it for settings of 10 or above:

```text
keyboards/abstract/ellipse/rev1/config.h:53:#define BACKLIGHT_LEVELS 15
keyboards/cannonkeys/satisfaction75/config.h:43:#define BACKLIGHT_LEVELS 24
keyboards/clueboard/66_hotswap/gen1/config.h:126:#define BACKLIGHT_LEVELS 10
keyboards/crkbd/keymaps/omgvee/config.h:45:#define BACKLIGHT_LEVELS 10
keyboards/handwired/co60/rev6/config.h:58:#define BACKLIGHT_LEVELS 24
keyboards/handwired/co60/rev7/config.h:60:#define BACKLIGHT_LEVELS 24
keyboards/jj40/config.h:36:#define BACKLIGHT_LEVELS 12
keyboards/jj40/keymaps/like_jis/config.h:45:#define BACKLIGHT_LEVELS 15
keyboards/jj4x4/config.h:39:#define BACKLIGHT_LEVELS 12
keyboards/jj50/config.h:39:#define BACKLIGHT_LEVELS 12
keyboards/keebio/iris/keymaps/transmogrified/config.h:46:#define BACKLIGHT_LEVELS 125
keyboards/lfkeyboards/lfk87/config.h:56:#define BACKLIGHT_LEVELS 10
keyboards/lfkeyboards/mini1800/config.h:43:#define BACKLIGHT_LEVELS 10
keyboards/meira/config.h:39:#define BACKLIGHT_LEVELS 10
keyboards/novelpad/config.h:56:#define BACKLIGHT_LEVELS 10
keyboards/planck/keymaps/dodger/config.h:56:#define BACKLIGHT_LEVELS 15
keyboards/satan/keymaps/olligranlund_iso/config.h:3:#define BACKLIGHT_LEVELS 10
keyboards/staryu/config.h:67:#define BACKLIGHT_LEVELS 10
keyboards/treasure/type9/config.h:53:#define BACKLIGHT_LEVELS 10
keyboards/xd84/config.h:51:#define BACKLIGHT_LEVELS 10
keyboards/xd96/config.h:51:#define BACKLIGHT_LEVELS 10
keyboards/ymd75/config.h:39:#define BACKLIGHT_LEVELS 12
keyboards/ymd96/config.h:39:#define BACKLIGHT_LEVELS 12
keyboards/ymd96/keymaps/AnthonyWharton/config.h:41:#define BACKLIGHT_LEVELS 12
keyboards/ymdk_np21/config.h:37:#define BACKLIGHT_LEVELS 12
layouts/community/ortho_4x12/mindsound/config.h:30:#define BACKLIGHT_LEVELS 15
```

`keyboards/keebio/iris/keymaps/transmogrified/config.h` is a massive outlier at `125`; excluding that result, the highest value in the repo is `24` (three instances), which would require 5 bits.

Reply:
Well, the docs say that 15 is the max..... 
https://docs.qmk.fm/#/feature_backlight?id=configuration

Reply:
I'm not sure where that number comes from - I haven't actually tested more than that but I suspect it will work. 15 is probably well on its way to being indistinguishable so might as well properly limit it to that.

As for moving the bit to the end - IMO having your backlight slightly dimmed is a small price to pay for neatness. The audio struct, for example, also places the level at the end, with the `enable` bit at the start and `clicky` in between.

Reply:
> I haven't actually tested more than that but I suspect it will work.

Just tested this. Backlight levels above 15 *will work*. I tried setting my `BACKLIGHT_LEVELS` to 25 and reflashed my boards. The differences in levels were subtle, but I could see the increases/decreases from one level to the next.

Reply:
So would you say 15 is a reasonable limit then?

Reply:
Yes, I think 15 is a reasonable limit.

Reply:
Looking at the code, 255 is the maximum possible, and there are no checks to enforce that.  

That may be something we want to add in the future.